backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/TryBackToAllSnapshotsOfVmCommand.java

```diff
@@ -211,14 +211,15 @@ private void restoreVmConfigFromSnapshot() {
             activeSnapshotLeaseDomainId = getVm().getStaticData().getLeaseStorageDomainId();
         }
 
+        Snapshot snapshot = snapshotDao.get(getVm().getId(), SnapshotType.ACTIVE);
         getSnapshotsManager().attemptToRestoreVmConfigurationFromSnapshot(getVm(),
                 getDstSnapshot(),
-                snapshotDao.getId(getVm().getId(), SnapshotType.ACTIVE),
+                snapshot.getId(),
                 getFilteredImagesToPreview(),
                 getCompensationContext(),
                 getCurrentUser(),
                 new VmInterfaceManager(getMacPool()),
-                getParameters().isRestoreMemory());
+                snapshot.getMemoryDiskId() != null);
```

Comment:
@ahadas Isn't it better to use the following instead?
```suggestion
            snapshot.containsMemory()
```
https://github.com/oVirt/ovirt-engine/blob/7fc4231be2880bf450fa62d3ee3dd44c5c75e7fb/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/Snapshot.java#L232-L234

Reply:
yes, but I can't merge this suggestion easily now - feel free to post another PR

Reply:
@ahadas done: https://github.com/oVirt/ovirt-engine/pull/667